mlflow/pyfunc/__init__.py

```diff
@@ -415,6 +415,43 @@ def predict(self, data: PyFuncInput) -> PyFuncOutput:
     def unwrap_python_model(self):
         """
         Unwrap the underlying Python model object.
+
+        This method is useful for accessing custom model functions, while still being able to
+        leverage the MLflow designed workflow through the `predict()` method.
+
+        .. code-block:: python
+            :caption: Example
+
+            # define a custom model
+            class MyModel(pyfunc.PythonModel):
+                def predict(self, context, model_input):
+                    self.my_custom_function(model_input)
+
+                def my_custom_function(self, model_input):
+                    # do something with the model input
+
+                def my_other_custom_function(self, *args, **kwargs):
+                    # do something else
+
+            # save the model
+            my_model = MyModel()
+            pyfunc.log_model(artifact_path="model", python_model=my_model)
+
+            # Register the model, transition to desired stage, do whatever else you want to do
```

Comment:
Do we need to register the model? I think we can remove this to simplify the example.

Reply:
Ideally, an example should be runnable without any changes. Here's an example

```python
print(a)
```

You can't run this code as it is because `a` is undefined. You need to define it by yourself, which is inconvenient.

```python
a = "b"
print(a)
```

You can run this code without any changes.